### PR TITLE
changed widget button into a div with a role of button so modal won't open when it shouldn't when enter is pressed

### PR DESCRIPTION
**Pull request by andydubois**

### CODE REVIEW

**Comment by gopida** [comment]
Few months ago, we changed from `div` to `button` for accessibility.  So couple things that comes to my mind is,  this may,

- Need `tabIndex` (to focus)
- Need `onClick` (to open modal) 
- Need `onKeyDown` (to open modal when enter or spacebar is pressed)

**Comment by aholdahl** [comment]
Watched local testing in screenshare, pending Andy's confirmation on Accessibility check (screenreader and at least 1 good scanner [IBM recommended]).

**Comment by andydubois** [comment]
> Few months ago, we changed from `div` to `button` for accessibility. So couple things that comes to my mind is, this may,
> 
> * Need `tabIndex` (to focus)
> * Need `onClick` (to open modal)
> * Need `onKeyDown` (to open modal when enter or spacebar is pressed)

I verified with Screen reader, while tabbing through the page the focus still goes to the Sezzle Widget in the logical order.  The modal also opened as expected when the enter key was pressed so OnClick and onKeyDown are not needed.

**Review by aholdahl — DISMISSED** [review]

**Comment by aholdahl** [comment]
Couple things: 1) add back the `tabindex="0"` to `.sezzle-modal-link`, just not to the `.sezzle-shopify-info-button`.
2) when using `Enter` to close the modal, it seems to be closing and re-opening rapidly...

**Review by aholdahl — APPROVED** [review]

File: src/modal.js [hunk 1]
```diff
@@ -33,35 +34,53 @@ class Modal {
...
+  }
+}
+
+  }
+
+  _sezzleAddKeyEvent(renderingInfo) {
+    const sezzleModalLinks = document.getElementsByClassName('sezzle-modal-link');
+    Array.prototype.forEach.call(sezzleModalLinks, (modalLink) => {
+      modalLink.addEventListener('keydown', (event) => {
+        if (event.key === "Enter" || event.key === "Space") {
```
Comment (gopida) [inline src/modal.js:73]: To act on `space` it should either be `event.code === "Space"` or `event.key === " "`

**Review by gopida — APPROVED** [review]

**Pull request merged**